app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -144,6 +144,12 @@ def perform_delete_in_cloud
       logger.warn("VMT with #{id_at_site} does not exist - continuing")
     end
 
+    # Returns the hourly cost for this template assuming a given VM flavor
+    def get_hourly_cost_for(flavor)
+      incarnation = flavor.flavor_os_families.find_by(os_family: appliance_type.os_family)
+      incarnation && incarnation.hourly_cost
+    end
```

Comment:
@mkasztelnik I have given this quite a lot of thought and IMHO no sensible "default value" can be returned in this context. Think about it: we are asking for the price of something which does not exist. What should the answer be? 0? 65535? +infinity? ;) Worse yet, if we return a numerical value, the client will later expect this exact cost to be applied when billing the appliance, complicating our model and also playing havoc with optimizer logic:

- How much does this cost?
- $1000.
- Okay, I want to buy it!
- Uh... well, you see... perhaps I could interest you in this other thing instead...? ;)

Realistically speaking, we have a choice between:
1. returning nil (which is what currently happens)
2. throwing an exception (which is a reasonable alternative)

In both cases the ball is in the client's court. If you feel strongly about choosing (2) over (1) then I will change the method accordingly.